packages/react-relay/__tests__/ReactRelayQueryRenderer-test.js

```diff
@@ -1602,6 +1602,68 @@ describe('ReactRelayQueryRenderer', () => {
 
       expect(render.mock.calls).toHaveLength(1);
     });
+    it('check GC with single pending request', () => {
+
+      const fetch = async () => response;
+      store = new Store(new RecordSource());
+      environment = new Environment({
+        network: Network.create(fetch),
+        store,
+      });
+
+      const renderer = ReactTestRenderer.create(
+          <ReactRelayQueryRenderer
+            environment={environment}
+            query={TestQuery}
+            render={render}
+            variables={variables}
+          />,
+      );
+
+      expect(store._roots.size).toEqual(1);
+      renderer.unmount();
+      expect(store._roots.size).toEqual(0);
+    });
+    it('check GC with multiple pending requests', () => {
```

Comment:
same question here

Reply:
same answer here but in a context where there are more references selected to be dispose. (this test shows how the previous version of react-relay did not delete all the referenced roots in the query renderer lifecycle)